# Introduction

Blood transfusion saves lives - from replacing lost blood during major surgery or a serious injury to treating various illnesses and blood disorders. Ensuring that there's enough blood in supply whenever needed is a serious challenge for the health professionals. According to WebMD,"about 5 million Americans need a blood transfusion every year". Blood transfusions are a crucial aspect of modern healthcare, with millions of units of blood transfused worldwide every year. However, the supply of blood is dependent on donors, and maintaining a stable supply of blood for transfusions is a challenging task. One of the significant challenges in blood donation is ensuring that donors come back to donate regularly, which helps to maintain a stable supply of blood.

In this capstone project, we aim to develop a predictive model that can help blood banks to identify donors who are likely to come back to donate blood based on a limited number of attributes. The model will leverage machine learning algorithms to analyze the data from previous blood donations and predict the likelihood of donors to return for subsequent donations.

The model will be developed using data obtained from a rom a mobile blood donation vehicle in Taiwan, including information on the time since last donation, number of donation, volume of donated blood, and total month past since first donation of donors. The Blood Transfusion Service Center drives to different universities and collects blood as part of a blood drive. The developed model will be evaluated using various metrics to determine its effectiveness in predicting future donor behavior.

The proposed predictive model has the potential to assist blood banks in increasing the number of repeat donors and maintaining a stable supply of blood. Moreover, this model can provide valuable insights into the factors that influence donor behavior, which can help blood banks to create targeted campaigns to encourage more people to donate blood regularly.

Overall, this capstone project aims to provide a data-driven approach to address the challenge of maintaining a stable supply of blood for transfusions by predicting donor behavior and helping blood banks to create targeted campaigns to increase repeat blood donations.

# Dataset

'Months since Last Donation': Itrepresents the number of months that have passed since the donor's last blood donation. This information can be useful in identifying donors who have not donated blood in a long time and may need to be encouraged to donate again.

'Number of Donations': This represents the total number of blood donations made by the donor. This information can be useful in identifying donors who have made multiple donations and may be more likely to donate again in the future.

'Total Volume Donated (c.c.)': This column represents the total volume of blood donated by the donor, measured in cubic centimeters. This information can be useful in identifying donors who have made significant contributions to the blood bank and may be more committed to donating blood.

'Months since First Donation': This feature represents the number of months that have passed since the donor's first blood donation. This information can be useful in identifying donors who have been donating blood for a long time and may be more likely to continue doing so.

'Made Donation in March 2007': It represents whether or not the donor made a blood donation in March 2007. It is target column in the dataset. This information can be useful in identifying donors who have recently donated blood and may be more likely to donate again in the near future.

Overall, this dataset appears to contain relevant information related to blood donation behavior, such as frequency and recency of donations, total volume donated, and whether or not the donor made a donation in a specific month. By analyzing this data and developing a predictive model, it may be possible to identify donors who are more likely to donate again and encourage them to make repeat donations, ultimately helping to maintain a stable supply of blood for transfusions.


# Domain Analysis

#### Blood Donation Industry:

Blood transfusions are a crucial aspect of modern healthcare, and millions of units of blood transfused worldwide every year. However, the supply of blood is dependent on donors, and maintaining a stable supply of blood for transfusions is a challenging task. This challenge is due to various factors, including the lack of awareness about blood donation, cultural barriers, and a lack of regular and repeat donors. According to the World Health Organization, blood donation by 1% of the population is usually sufficient to meet a country's basic blood needs. In most countries, however, voluntary unpaid donors account for less than 50% of blood donations, resulting in an unsteady supply of blood for transfusions.

#### Donor Behavior:

Understanding donor behavior is crucial in identifying ways to maintain a stable blood supply. Various factors influence whether or not someone donates blood, including age, gender, socio-economic status, and personal motivations. A study conducted by Adereti et al. (2019) found that the most common reason for donating blood was to help others, followed by a personal experience with a friend or family member who required a transfusion. However, other factors such as inconvenience, fear of needles, and a lack of awareness about blood donation can discourage people from donating blood.

#### Blood Donation Campaigns:

Many organizations run blood donation campaigns to encourage people to donate blood. These campaigns often utilize a range of strategies, such as social media campaigns, posters, and incentives for donors. For example, a blood donation campaign run by the American Red Cross in 2020 provided donors with a $5 Amazon gift card as an incentive. Additionally, blood donation campaigns often target specific demographics or populations that are underrepresented in blood donation, such as young adults or ethnic minorities.

#### Data Analytics in Healthcare:

Data analytics is becoming an increasingly important tool in healthcare for improving patient outcomes and resource allocation. In the blood donation industry, data analytics can help to identify patterns in donor behavior, such as the factors that influence whether or not someone donates blood and the likelihood of a donor making a repeat donation. Furthermore, predictive modeling can help to identify donors who are more likely to donate again, allowing blood banks to focus their resources on encouraging repeat donations.

#### Predictive Modeling:

Predictive modeling is a technique used in data analytics to make predictions about future events based on historical data. In healthcare, predictive modeling has various applications, including predicting patient outcomes, identifying individuals at risk of disease, and forecasting healthcare demand. In the context of blood donation, predictive modeling can be used to identify donors who are more likely to donate again and create targeted campaigns to encourage repeat donations. Machine learning algorithms such as random forest, logistic regression, and decision trees are used to build predictive models from donor data.



# Problem Type and Performance Matric

Based on the given dataset, the blood donation project can be formulated as a binary classification problem. The goal is to predict whether a donor will donate blood in March 2007 (Made Donation in March 2007) based on their past donation behavior and demographic attributes. The target variable is binary, and the dataset contains both categorical and continuous features. Therefore, suitable machine learning algorithms for this problem include logistic regression, decision trees, random forest, support vector machines (SVMs), and neural networks.

When evaluating the performance of a binary classification model, several performance metrics can be used, depending on the specific needs of the project. Some of the common performance metrics used for binary classification problems include:

Accuracy: The proportion of correct predictions made by the model.

Precision: The proportion of true positives (correctly predicted positive class) among all predicted positives.

Recall: The proportion of true positives among all actual positive samples.

F1-score: The harmonic mean of precision and recall, which combines both metrics.

Area Under the ROC Curve (AUC-ROC): The area under the receiver operating characteristic (ROC) curve, which measures the trade-off between true positive rate and false positive rate.

The selection of a suitable performance metric depends on the specific needs and objectives of the project. However, in the case of the blood donation project, where the goal is to predict whether a donor will donate blood in March 2007, the most appropriate performance metrics to use would be precision, recall, and the F1-score.

Precision is important because it measures the proportion of correctly predicted positive class (i.e., donors who will donate in March 2007) among all predicted positives. In the context of the blood donation project, high precision indicates that the model is accurately identifying donors who are likely to donate, which is critical for effective donor targeting.

Recall is also essential in this problem because it measures the proportion of actual positive samples (i.e., donors who donated in March 2007) that were correctly identified by the model. High recall indicates that the model can identify a high percentage of donors who will donate in March 2007, which is important for maintaining a stable blood supply.

The F1-score is a harmonic mean of precision and recall, which combines both metrics. It provides a single measure of overall performance that is useful for comparing different models and selecting the best one.

While accuracy and AUC-ROC are also important performance metrics, they may not be the most appropriate for this particular problem. Accuracy is not always the best measure of performance in imbalanced datasets, where the number of positive and negative samples is not equal. In the blood donation dataset, the positive class (donors who will donate in March 2007) is likely to be less frequent than the negative class, making accuracy less informative. Similarly, AUC-ROC measures the trade-off between true positive rate and false positive rate, which may not be the most relevant metric for this problem where the focus is on identifying donors who will donate. Therefore, precision, recall, and the F1-score are the most appropriate performance metrics to use in the blood donation project.

# Importing Libraries

In [21]:
import pandas as pd
import numpy as np

In [22]:
donation= pd.read_csv('blood_don.csv')
donation.head()

Unnamed: 0  Months since Last Donation  Number of Donations  \
0         619                           2                   50   
1         664                           0                   13   
2         441                           1                   16   
3         160                           2                   20   
4         358                           1                   24   

   Total Volume Donated (c.c.)  Months since First Donation  \
0                        12500                           98   
1                         3250                           28   
2                         4000                           35   
3                         5000                           45   
4                         6000                           77   

   Made Donation in March 2007  
0                            1  
1                            1  
2                            1  
3                            1  
4                            0

In [23]:
donation.shape

(576, 6)

There are 576 rows and 6 columns in the dataset

In [24]:
donation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 576 entries, 0 to 575
Data columns (total 6 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   Unnamed: 0                   576 non-null    int64
 1   Months since Last Donation   576 non-null    int64
 2   Number of Donations          576 non-null    int64
 3   Total Volume Donated (c.c.)  576 non-null    int64
 4   Months since First Donation  576 non-null    int64
 5   Made Donation in March 2007  576 non-null    int64
dtypes: int64(6)
memory usage: 27.1 KB


- There are no missing values
- All values are numeric and of int64 type

In [25]:
donation.columns

Index(['Unnamed: 0', 'Months since Last Donation', 'Number of Donations',
       'Total Volume Donated (c.c.)', 'Months since First Donation',
       'Made Donation in March 2007'],
      dtype='object')